src/server/router.js

```diff
@@ -188,6 +218,11 @@ export class MockerRouter {
         const request = new MockerRequest(event, rule);
         const response = new MockerResponse(event);
 
+        // apply middleware
+        this._middleware.forEach((fn) => {
+          fn.call(event, request, response);
```

Comment:
I think it could be a good idea to allow middleware to be async (return a promise). It will open lot's of possibilities